src/dodal/devices/b16/detector.py

```diff
@@ -0,0 +1,34 @@
+from ophyd_async.epics.adcore import (
+    ADBaseController,
+    ADBaseIO,
+    ADTIFFWriter,
+    AreaDetector,
+)
+
+from dodal.common.beamlines.beamline_utils import get_path_provider
+from dodal.common.beamlines.device_helpers import CAM_SUFFIX, TIFF_SUFFIX
+
+
+class ConstantDeadTimeController(ADBaseController):
```

Comment:
Could: @coretl, is this something that should be in `ophyd-async` already? Feels like it might be generically useful